topo/node/ceos/ceos.go

```diff
@@ -329,15 +340,21 @@ func defaults(pb *tpb.Node) *tpb.Node {
 	return pb
 }
 
-func (n *Node) FixInterfaces() {
+func (n *Node) FixInterfaces() error {
 	for k, v := range n.Proto.Interfaces {
 		if !strings.HasPrefix(k, "eth") {
```

Comment:
a switch statement would be more clear here:

switch {
default:
  return fmt.Errorf("unrecognized interface name: %s", v.Name)
case !strings.HasPrefix(k, "eth"), ethIntfRe.MatchString(v.Name), mgmtIntfRe.MatchString(v.Name):
case v.Name == "":
  n.Proto.Interfaces[k].Name = fmt.Sprintf("Ethernet%s", strings.TrimPrefix(k, "eth"))
}
return nil

Reply:
+1